src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -348,10 +352,15 @@ void Complete()
                         {
                             result = System.Array.Empty<object>();
                         }
+                        else if (IsAssignableToGenericTaskType(task_type))
```

Comment:
Why is this lookup needed we want to support Task<T> only?

Reply:
In order to make the code trimming-compatible, a slight behavior change was made. The SetupJSContinuation method will only get the Result of Task{TResult} objects. If an object derives from the base Task class, and defines its own Result property, SetupJSContinuation will no longer respect that property.

This was done so the behavior remains consistent between trimmed and untrimmed applications.

Reply:
I don't think that's the behaviour we want. The idea is to support only handpicked types, certainly not anything user-defined.

@lewing could you confirm?

Reply:
I can change it to only supporting Task{TResult} directly. I just thought I’d preserve as much existing behavior as possible here.

Reply:
Actually, limiting this only to `Task<TResult>` directly won't work since we have our own built-in types that derive from `Task<TResult>`. For example:

https://github.com/dotnet/runtime/blob/8636dc8a76d687169db64033b71ec9f79d9ba4b5/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs#L271-L275

https://github.com/dotnet/runtime/blob/8636dc8a76d687169db64033b71ec9f79d9ba4b5/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs#L513

https://github.com/dotnet/runtime/blob/8636dc8a76d687169db64033b71ec9f79d9ba4b5/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs#L5867-L5872